<?xml version="1.0" encoding="UTF-8"?>
<!--This does not replace URDF, and is not an extension of URDF.
    This is a format for representing semantic information about the robot structure.
    A URDF file must exist for this robot as well, where the joints and the links that are referenced are defined
-->
<robot name="g1_29dof">

	<!--group name="base">
        <link name="pelvis"/>
    </group-->
    <!--group name="imu_sensors">
        <link name="imu_link"/>
    </group-->
    <group name="left_leg">
        <chain base_link="pelvis" tip_link="left_ankle_roll_link"/>
    </group>
    <group name="right_leg">
        <chain base_link="pelvis" tip_link="right_ankle_roll_link"/>
    </group>
    <group name="torso">
    	<chain base_link="pelvis" tip_link="torso_link"/>
    </group>
    <group name="left_arm">
        <chain base_link="torso_link" tip_link="left_wrist_yaw_link"/>
    </group>
    <group name="right_arm">
        <chain base_link="torso_link" tip_link="right_wrist_yaw_link"/>
    </group>
    
    <group name="chains">
        <group name="left_leg"/>
        <group name="right_leg"/>
        <group name="torso"/>
        <group name="left_arm"/>
        <group name="right_arm"/>
    </group>
    
    <group_state name="home" group="chains">
    	<joint name="left_hip_pitch_joint" value="-0.15"/>
    	<joint name="left_hip_roll_joint" value="0"/>
    	<joint name="left_hip_yaw_joint" value="0"/>
    	<joint name="left_knee_joint" value="0.45"/>
    	<joint name="left_ankle_pitch_joint" value="-0.3"/>
    	<joint name="left_ankle_roll_joint" value="0"/>
    	<joint name="right_hip_pitch_joint" value="-0.15"/>
    	<joint name="right_hip_roll_joint" value="0"/>
    	<joint name="right_hip_yaw_joint" value="0"/>
    	<joint name="right_knee_joint" value="0.45"/>
    	<joint name="right_ankle_pitch_joint" value="-0.3"/>
    	<joint name="right_ankle_roll_joint" value="0"/>
    	<joint name="waist_yaw_joint" value="0"/>
    	<joint name="waist_roll_joint" value="0"/>
    	<joint name="waist_pitch_joint" value="0."/>
    	<joint name="left_shoulder_pitch_joint" value="0"/>
    	<joint name="left_shoulder_roll_joint" value="0"/>
    	<joint name="left_shoulder_yaw_joint" value="0"/>
    	<joint name="left_elbow_joint" value="0."/>
    	<joint name="left_wrist_roll_joint" value="0"/>
    	<joint name="left_wrist_pitch_joint" value="0"/>
    	<joint name="left_wrist_yaw_joint" value="0"/>
    	<joint name="right_shoulder_pitch_joint" value="0"/>
    	<joint name="right_shoulder_roll_joint" value="0"/>
    	<joint name="right_shoulder_yaw_joint" value="0"/>
    	<joint name="right_elbow_joint" value="0."/>
    	<joint name="right_wrist_roll_joint" value="0"/>
    	<joint name="right_wrist_pitch_joint" value="0"/>
    	<joint name="right_wrist_yaw_joint" value="0"/>
    </group>


    <!--DISABLE COLLISIONS: By default it is assumed that any link of the robot could potentially come into collision with any other link in the robot. This tag disables collision checking between a specified pair of links. -->
    <disable_collisions link1="head_link" link2="left_hip_roll_link" reason="Never"/>
    <disable_collisions link1="head_link" link2="left_shoulder_pitch_link" reason="Never"/>
    <disable_collisions link1="head_link" link2="logo_link" reason="Never"/>
    <disable_collisions link1="head_link" link2="pelvis_contour_link" reason="Never"/>
    <disable_collisions link1="head_link" link2="right_hip_pitch_link" reason="Never"/>
    <disable_collisions link1="head_link" link2="right_hip_roll_link" reason="Never"/>
    <disable_collisions link1="head_link" link2="right_shoulder_pitch_link" reason="Never"/>
    <disable_collisions link1="head_link" link2="torso_link" reason="Adjacent"/>
    <disable_collisions link1="head_link" link2="waist_support_link" reason="Never"/>
    <disable_collisions link1="left_ankle_pitch_link" link2="left_ankle_roll_link" reason="Adjacent"/>
    <disable_collisions link1="left_ankle_pitch_link" link2="left_hip_roll_link" reason="Never"/>
    <disable_collisions link1="left_ankle_pitch_link" link2="left_hip_yaw_link" reason="Never"/>
    <disable_collisions link1="left_ankle_pitch_link" link2="left_knee_link" reason="Adjacent"/>
    <disable_collisions link1="left_ankle_roll_link" link2="left_hip_yaw_link" reason="Never"/>
    <disable_collisions link1="left_ankle_roll_link" link2="left_knee_link" reason="Never"/>
    <disable_collisions link1="left_ankle_roll_link" link2="right_ankle_roll_link" reason="Never"/>
    <disable_collisions link1="left_elbow_link" link2="left_shoulder_pitch_link" reason="Never"/>
    <disable_collisions link1="left_elbow_link" link2="left_shoulder_roll_link" reason="Never"/>
    <disable_collisions link1="left_elbow_link" link2="left_shoulder_yaw_link" reason="Adjacent"/>
    <disable_collisions link1="left_elbow_link" link2="left_wrist_pitch_link" reason="Never"/>
    <disable_collisions link1="left_elbow_link" link2="left_wrist_roll_link" reason="Adjacent"/>
    <disable_collisions link1="left_elbow_link" link2="left_wrist_yaw_link" reason="Never"/>
    <disable_collisions link1="left_hip_roll_link" link2="left_hip_yaw_link" reason="Adjacent"/>
    <disable_collisions link1="left_hip_roll_link" link2="left_knee_link" reason="Never"/>
    <disable_collisions link1="left_hip_roll_link" link2="left_shoulder_pitch_link" reason="Never"/>
    <disable_collisions link1="left_hip_roll_link" link2="left_shoulder_roll_link" reason="Never"/>
    <disable_collisions link1="left_hip_roll_link" link2="logo_link" reason="Never"/>
    <disable_collisions link1="left_hip_roll_link" link2="pelvis_contour_link" reason="Adjacent"/>
    <disable_collisions link1="left_hip_roll_link" link2="right_hip_pitch_link" reason="Adjacent"/>
    <disable_collisions link1="left_hip_roll_link" link2="right_hip_roll_link" reason="Never"/>
    <disable_collisions link1="left_hip_roll_link" link2="right_hip_yaw_link" reason="Never"/>
    <disable_collisions link1="left_hip_roll_link" link2="right_shoulder_pitch_link" reason="Never"/>
    <disable_collisions link1="left_hip_roll_link" link2="right_shoulder_roll_link" reason="Never"/>
    <disable_collisions link1="left_hip_roll_link" link2="torso_link" reason="Adjacent"/>
    <disable_collisions link1="left_hip_yaw_link" link2="left_knee_link" reason="Adjacent"/>
    <disable_collisions link1="left_hip_yaw_link" link2="pelvis_contour_link" reason="Never"/>
    <disable_collisions link1="left_hip_yaw_link" link2="right_hip_pitch_link" reason="Never"/>
    <disable_collisions link1="left_hip_yaw_link" link2="right_hip_roll_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_pitch_link" link2="left_shoulder_roll_link" reason="Adjacent"/>
    <disable_collisions link1="left_shoulder_pitch_link" link2="left_shoulder_yaw_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_pitch_link" link2="left_wrist_roll_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_pitch_link" link2="logo_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_pitch_link" link2="pelvis_contour_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_pitch_link" link2="right_hip_pitch_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_pitch_link" link2="right_hip_roll_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_pitch_link" link2="right_shoulder_pitch_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_pitch_link" link2="right_shoulder_roll_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_pitch_link" link2="right_shoulder_yaw_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_pitch_link" link2="torso_link" reason="Adjacent"/>
    <disable_collisions link1="left_shoulder_pitch_link" link2="waist_support_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_roll_link" link2="left_shoulder_yaw_link" reason="Adjacent"/>
    <disable_collisions link1="left_shoulder_roll_link" link2="left_wrist_roll_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_roll_link" link2="logo_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_roll_link" link2="pelvis_contour_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_roll_link" link2="right_hip_pitch_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_roll_link" link2="right_hip_roll_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_roll_link" link2="right_shoulder_pitch_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_roll_link" link2="right_shoulder_roll_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_roll_link" link2="waist_support_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_yaw_link" link2="left_wrist_pitch_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_yaw_link" link2="left_wrist_roll_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_yaw_link" link2="left_wrist_yaw_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_yaw_link" link2="pelvis_contour_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_yaw_link" link2="right_hip_pitch_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_yaw_link" link2="right_shoulder_pitch_link" reason="Never"/>
    <disable_collisions link1="left_shoulder_yaw_link" link2="waist_support_link" reason="Never"/>
    <disable_collisions link1="left_wrist_pitch_link" link2="left_wrist_roll_link" reason="Adjacent"/>
    <disable_collisions link1="left_wrist_pitch_link" link2="left_wrist_yaw_link" reason="Adjacent"/>
    <disable_collisions link1="left_wrist_roll_link" link2="left_wrist_yaw_link" reason="Never"/>
    <disable_collisions link1="logo_link" link2="pelvis_contour_link" reason="Never"/>
    <disable_collisions link1="logo_link" link2="right_hip_pitch_link" reason="Never"/>
    <disable_collisions link1="logo_link" link2="right_hip_roll_link" reason="Never"/>
    <disable_collisions link1="logo_link" link2="right_shoulder_pitch_link" reason="Never"/>
    <disable_collisions link1="logo_link" link2="right_shoulder_roll_link" reason="Never"/>
    <disable_collisions link1="logo_link" link2="torso_link" reason="Adjacent"/>
    <disable_collisions link1="logo_link" link2="waist_support_link" reason="Never"/>
    <disable_collisions link1="pelvis_contour_link" link2="right_hip_pitch_link" reason="Adjacent"/>
    <disable_collisions link1="pelvis_contour_link" link2="right_hip_yaw_link" reason="Never"/>
    <disable_collisions link1="pelvis_contour_link" link2="right_shoulder_pitch_link" reason="Never"/>
    <disable_collisions link1="pelvis_contour_link" link2="right_shoulder_roll_link" reason="Never"/>
    <disable_collisions link1="pelvis_contour_link" link2="right_shoulder_yaw_link" reason="Never"/>
    <disable_collisions link1="pelvis_contour_link" link2="torso_link" reason="Adjacent"/>
    <disable_collisions link1="pelvis_contour_link" link2="waist_support_link" reason="Never"/>
    <disable_collisions link1="right_ankle_pitch_link" link2="right_ankle_roll_link" reason="Adjacent"/>
    <disable_collisions link1="right_ankle_pitch_link" link2="right_hip_roll_link" reason="Never"/>
    <disable_collisions link1="right_ankle_pitch_link" link2="right_hip_yaw_link" reason="Never"/>
    <disable_collisions link1="right_ankle_pitch_link" link2="right_knee_link" reason="Adjacent"/>
    <disable_collisions link1="right_ankle_roll_link" link2="right_hip_yaw_link" reason="Never"/>
    <disable_collisions link1="right_ankle_roll_link" link2="right_knee_link" reason="Never"/>
    <disable_collisions link1="right_elbow_link" link2="right_shoulder_pitch_link" reason="Never"/>
    <disable_collisions link1="right_elbow_link" link2="right_shoulder_roll_link" reason="Never"/>
    <disable_collisions link1="right_elbow_link" link2="right_shoulder_yaw_link" reason="Adjacent"/>
    <disable_collisions link1="right_elbow_link" link2="right_wrist_pitch_link" reason="Never"/>
    <disable_collisions link1="right_elbow_link" link2="right_wrist_roll_link" reason="Adjacent"/>
    <disable_collisions link1="right_elbow_link" link2="right_wrist_yaw_link" reason="Never"/>
    <disable_collisions link1="right_hip_pitch_link" link2="right_hip_roll_link" reason="Adjacent"/>
    <disable_collisions link1="right_hip_pitch_link" link2="right_hip_yaw_link" reason="Never"/>
    <disable_collisions link1="right_hip_pitch_link" link2="right_knee_link" reason="Never"/>
    <disable_collisions link1="right_hip_pitch_link" link2="right_shoulder_pitch_link" reason="Never"/>
    <disable_collisions link1="right_hip_pitch_link" link2="right_shoulder_roll_link" reason="Never"/>
    <disable_collisions link1="right_hip_pitch_link" link2="right_shoulder_yaw_link" reason="Never"/>
    <disable_collisions link1="right_hip_pitch_link" link2="torso_link" reason="Adjacent"/>
    <disable_collisions link1="right_hip_pitch_link" link2="waist_support_link" reason="Never"/>
    <disable_collisions link1="right_hip_roll_link" link2="right_hip_yaw_link" reason="Adjacent"/>
    <disable_collisions link1="right_hip_roll_link" link2="right_knee_link" reason="Never"/>
    <disable_collisions link1="right_hip_roll_link" link2="right_shoulder_pitch_link" reason="Never"/>
    <disable_collisions link1="right_hip_roll_link" link2="right_shoulder_roll_link" reason="Never"/>
    <disable_collisions link1="right_hip_yaw_link" link2="right_knee_link" reason="Adjacent"/>
    <disable_collisions link1="right_shoulder_pitch_link" link2="right_shoulder_roll_link" reason="Adjacent"/>
    <disable_collisions link1="right_shoulder_pitch_link" link2="right_shoulder_yaw_link" reason="Never"/>
    <disable_collisions link1="right_shoulder_pitch_link" link2="right_wrist_roll_link" reason="Never"/>
    <disable_collisions link1="right_shoulder_pitch_link" link2="torso_link" reason="Adjacent"/>
    <disable_collisions link1="right_shoulder_pitch_link" link2="waist_support_link" reason="Never"/>
    <disable_collisions link1="right_shoulder_roll_link" link2="right_shoulder_yaw_link" reason="Adjacent"/>
    <disable_collisions link1="right_shoulder_roll_link" link2="right_wrist_roll_link" reason="Never"/>
    <disable_collisions link1="right_shoulder_roll_link" link2="waist_support_link" reason="Never"/>
    <disable_collisions link1="right_shoulder_yaw_link" link2="right_wrist_pitch_link" reason="Never"/>
    <disable_collisions link1="right_shoulder_yaw_link" link2="right_wrist_roll_link" reason="Never"/>
    <disable_collisions link1="right_shoulder_yaw_link" link2="right_wrist_yaw_link" reason="Never"/>
    <disable_collisions link1="right_shoulder_yaw_link" link2="waist_support_link" reason="Never"/>
    <disable_collisions link1="right_wrist_pitch_link" link2="right_wrist_roll_link" reason="Adjacent"/>
    <disable_collisions link1="right_wrist_pitch_link" link2="right_wrist_yaw_link" reason="Adjacent"/>
    <disable_collisions link1="right_wrist_roll_link" link2="right_wrist_yaw_link" reason="Never"/>
    <disable_collisions link1="torso_link" link2="waist_support_link" reason="Adjacent"/>
</robot>
